rust/src/lightning.rs

```diff
@@ -558,77 +558,7 @@ pub async fn run_ldk_server(
 
     tracing::info!("Listening to 0.0.0.0:{listening_port}");
 
-    let event_handler = {
-        let channel_manager = system.channel_manager.clone();
-        let wallet = system.wallet.clone();
-        let network_graph = system.network_graph.clone();
-        let keys_manager = system.keys_manager.clone();
-        let inbound_payments = system.inbound_payments.clone();
-        let outbound_payments = system.outbound_payments.clone();
-        let network = system.network;
-        {
-            move |event: &Event| {
-                block_on(handle_ldk_events(
-                    channel_manager.clone(),
-                    wallet.clone(),
-                    &network_graph,
-                    keys_manager.clone(),
-                    inbound_payments.clone(),
-                    outbound_payments.clone(),
-                    network,
-                    event,
-                ));
-            }
-        }
-    };
-
-    let scorer_path = format!("{ldk_data_dir}/scorer");
-    let scorer = Arc::new(Mutex::new(disk::read_scorer(
-        Path::new(&scorer_path),
-        Arc::clone(&system.network_graph),
-        Arc::clone(&system.logger),
-    )));
-
-    let router = DefaultRouter::new(
-        system.network_graph.clone(),
-        system.logger.clone(),
-        system.keys_manager.get_secure_random_bytes(),
-        scorer.clone(),
-    );
-    let invoice_payer = Arc::new(InvoicePayer::new(
-        system.channel_manager.clone(),
-        router,
-        system.logger.clone(),
-        event_handler,
-        payment::Retry::Timeout(Duration::from_secs(10)),
-    ));
-
-    let background_processor = BackgroundProcessor::start(
-        system.persister.clone(),
-        invoice_payer.clone(),
-        system.chain_monitor.clone(),
-        system.channel_manager.clone(),
-        GossipSync::p2p(system.gossip_sync.clone()),
-        system.peer_manager.clone(),
-        system.logger.clone(),
-        Some(scorer),
-    );
-    let peer_data_path = format!("{ldk_data_dir}/channel_peer_data");
-    let mut info = disk::read_channel_peer_data(Path::new(&peer_data_path))?;
-    for (pubkey, peer_addr) in info.drain() {
-        for chan_info in system.channel_manager.list_channels() {
-            if pubkey == chan_info.counterparty.node_id {
-                block_on(async {
-                    let _ = connect_peer_if_necessary(
-                        &PeerInfo { pubkey, peer_addr },
-                        system.peer_manager.clone(),
-                    )
-                    .await;
-                });
-            }
-        }
-    }
-    tracing::info!("Lightning network started");
+    let background_processor = run_ldk(system).await?;
```

Comment:
We could extract the listening part into a separate function and have two calls on the server, one for starting the listening and then just `run_ldk`. The `tcp_handle is not used in `run_ldk`.
Not important enough to fix now though :)